### PR TITLE
Refactor files trashbin app

### PR DESCRIPTION
**Pull request by danialRahimy**
## Summary

I did these items in a single file `apps/files_trashbin/lib/Trashbin.php` for two reasons:
1. Pull request hasn't been a lot of changes.
2. I was not sure about some changes.

- Return early
- Change the const name for more readability
- Add functions' return type
- Update doc types

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files_trashbin/lib/Trashbin.php [hunk 1]
```diff
@@ -227,16 +246,21 @@ private static function copyFilesToUser($sourcePath, $owner, $targetPath, $user,
...
 	 * @param bool $ownerOnly delete for owner only (if file gets moved out of a shared folder)
-	 *
 	 * @return bool
+	 * @throws Exception
+	 * @throws CopyRecursiveException
+	 * @throws InvalidPathException
+	 * @throws LockedException
+	 * @throws NoUserException
+	 * @throws QueryException
+	 * @throws ContainerExceptionInterface
```
Comment (nickvergessen) [inline apps/files_trashbin/lib/Trashbin.php:261]: I'm very much against these.
They are not expected and not part of the API. If they happen it's not a user / API consumer fault, but basically the whole nextcloud is offline/broken.

Comment (danialRahimy) [reply apps/files_trashbin/lib/Trashbin.php:261]: You're right, and I appreciate your input.
I'll revert the changes to their previous state.

File: apps/files_trashbin/lib/Trashbin.php [hunk 2]
```diff
@@ -66,15 +68,15 @@
...
+	public const DEFAULT_MAX_SIZE = 50;
 
 	/**
 	 * Ensure we don't need to scan the file during the move to trash
 	 * by triggering the scan in the pre-hook
 	 *
 	 * @param array $params
 	 */
-	public static function ensureFileScannedHook($params) {
+	public static function ensureFileScannedHook($params): void {
```
Comment (Fenn-CS) [inline apps/files_trashbin/lib/Trashbin.php:79]:
```suggestion
	public static function ensureFileScannedHook(array $params): void {
```

File: apps/files_trashbin/lib/Trashbin.php [hunk 3]
```diff
@@ -88,9 +90,8 @@ public static function ensureFileScannedHook($params) {
 	 *
 	 * @param string $filename
 	 * @return array
-	 * @throws \OC\User\NoUserException
 	 */
-	public static function getUidAndFilename($filename) {
+	public static function getUidAndFilename($filename): array {
```
Comment (Fenn-CS) [inline apps/files_trashbin/lib/Trashbin.php:94]:
```suggestion
	public static function getUidAndFilename(string $filename): array {
```

File: apps/files_trashbin/lib/Trashbin.php [hunk 4]
```diff
@@ -122,7 +123,7 @@ public static function getUidAndFilename($filename) {
 	 * @param string $user
 	 * @return array (filename => array (timestamp => original location))
 	 */
-	public static function getLocations($user) {
+	public static function getLocations($user): array {
```
Comment (Fenn-CS) [inline apps/files_trashbin/lib/Trashbin.php:126]:
```suggestion
	public static function getLocations(string $user): array {
```

File: apps/files_trashbin/lib/Trashbin.php [hunk 5]
```diff
@@ -146,9 +147,9 @@ public static function getLocations($user) {
 	 * @param string $user
 	 * @param string $filename
 	 * @param string $timestamp
-	 * @return string original location
+	 * @return bool|string original location
 	 */
-	public static function getLocation($user, $filename, $timestamp) {
+	public static function getLocation($user, $filename, $timestamp): bool|string {
```
Comment (Fenn-CS) [inline apps/files_trashbin/lib/Trashbin.php:152]:
```suggestion
	public static function getLocation(string $user, string $filename, string $timestamp): bool|string {
```

File: apps/files_trashbin/lib/Trashbin.php [hunk 6]
```diff
@@ -167,7 +168,10 @@ public static function getLocation($user, $filename, $timestamp) {
 		}
 	}
 
-	private static function setUpTrash($user) {
+	/**
+	 * @param string $user
+	 */
+	private static function setUpTrash($user): void {
```
Comment (Fenn-CS) [inline apps/files_trashbin/lib/Trashbin.php:174]:
```suggestion
	private static function setUpTrash(string $user): void {
```

File: apps/files_trashbin/lib/Trashbin.php [hunk 7]
```diff
@@ -183,17 +187,18 @@ private static function setUpTrash($user) {
...
 	 * @param string $owner
 	 * @param string $targetPath
-	 * @param $user
+	 * @param string $user
 	 * @param int $timestamp
+	 * @throws CopyRecursiveException
+	 * @throws InvalidPathException
 	 */
-	private static function copyFilesToUser($sourcePath, $owner, $targetPath, $user, $timestamp) {
+	private static function copyFilesToUser($sourcePath, $owner, $targetPath, $user, $timestamp): void {
```
Comment (Fenn-CS) [inline apps/files_trashbin/lib/Trashbin.php:201]:
```suggestion
	private static function copyFilesToUser(string $sourcePath, string $owner, string $targetPath, string $user, int $timestamp): void {
```

**Review by nfebe — CHANGES_REQUESTED** [review]
Since this is just a refactor and no main functional change you can just use build in type hinting for PHP... I left a few examples but not all, most of the functions can use built in type hinting and that would allow you to delete the comment annotations at the top (leaving behind just the exceptions)

**Pull request closed without merging**